# Intro to programming with Python (Notes)
* Python Operators:
  * **Arithmetic**: Addition `+`, subtraction `-`, multiplication `*`, division `/`, ground division `//`, remainder `%`, exponent `**`.

  * **Logic**: `and`, `or`, `not`.
 * **Conditional**: Equal `==`, greater than `>`, less than `<`, not equal `!=`, greater than or equal `>=`, less than or equal `<=`.

###Indented texts are from: "https://jovian.ai/learn/data-analysis-with-python-zero-to-pandas/lesson/lesson-1-introduction-to-programming-with-python".

>Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the //, %, and ** operators are used for?



In [1]:
2 + 3 + 9

14

In [2]:
99 - 73

26

In [3]:
23.54 * -1432

-33709.28

In [4]:
100 / 7

14.285714285714286

In [5]:
100 // 7

14

In [6]:
100 % 7

2

In [7]:
5 ** 3

125

>As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [8]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

>Python supports the following arithmetic operators:




| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


>Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 


In [9]:
#1. The population of a town is 198568. Out of them 45312 are men and 35678 are women. Find the number of children in the town.
town = 198568
men = 45312
women = 35678
children = town - (men + women)
children

117578

In [10]:
#2. A shopkeeper has 2425 boxes of 24 pencils each. How many pencils do all the boxes have in all?
boxes = 2425
pencils = 24
total = 2425 * 24
total

58200

In [11]:
#3. Linda bought a coat for $2265 and a saree for $2150. She gave $5000 to the shopkeeper. How much money did the shopkeeper return to her?
coat = 2265
saree = 2150
gave = 5000
returns = gave - (coat + saree)
returns

585

In [12]:
#4. The cost of 21 TV sets is $95844. Find the cost of one TV set.
total = 95844
sets = 21
oneTV = total / 21
oneTV

4564.0

In [13]:
#5. A factory produces 24532 bulbs in a month. What is its annual production?
month_production = 24532
months_in_a_year = 12
total_production = month_production * months_in_a_year
print(total_production,"bulbs a year.")

294384 bulbs a year.


In [15]:
"""6. There are 145968 bags of sugar, 236487 bags of wheat and some bags of rice in a godown. It the total number of bags in 
the godown is 450000, find the number of bags of rice."""
sugar_bags = 145968
wheat_bags = 236487
total_godown = 450000

rice_bags = total_godown - (sugar_bags + wheat_bags)
rice_bags


67545

In [16]:
"""7. A factory manufactured 483685 toys in three weeks. 
The production in first week was 146345 toys and in second week 138152 toys. Find the production in the third week."""
total = 483685
first_week = 146345
second_week = 138152
third_week = total - (146345 + 138152)
third_week

199188

In [17]:
#8. The cost of a sofa set is $9372. How much will 124 such sofa sets cost?
one_set = 9372
total_sets = 124
total_cost = one_set * total_sets
total_cost

1162128

In [19]:
#9. There are 86 rooms in a school. 4356 students study there. Equal number of students sits in each room?
number_of_rooms = 86
students = 4356
students_per_room = students // number_of_rooms
students_per_room

50

In [20]:
"""10. 1575 students of a school want to go Agra by bus. 
If one bus can carry 75 students, how many buses are required to carry all the students?"""
school_students = 1575
bus_can_carry = 75
number_of_bus_needed = school_students // bus_can_carry
number_of_bus_needed

21

In [21]:
#11. The cost of a radio set is 1475. What is the cost of 35 such radio sets?
one_radio_set = 1475
needed = 35
total_price = one_radio_set * needed
total_price

51625

In [22]:
"""12. In an election, 52496 people voted for Ron, 44929 people for Jhon and 36824 people for Mike in a town. 
If everyone voted in the town, what is the total number of voters?"""
ron = 52496
jhon = 44929
mike = 36824
total_voters = ron + jhon + mike
total_voters

134249

In [28]:
"""13. Maria bought 96 toys priced equally for $12960. The amount of $1015 is still left with her.
Find the cost of each toy and the amount she had."""
number_of_toys = 96
priced = 12960
amount_left = 1015
each_priced = priced // number_of_toys
total_she_had = priced + amount_left
each_priced, total_she_had

(135, 13975)

## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [29]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

*Plane costs* = $5000

*Signed up people* = 29

*Ticket* = $200 each

In [30]:
#defining the costs
ticket = 200
people = 29
plane_cost = 5000

#Total tickets recieved
money_recieved_from_tickets = ticket * people

#Ticket - plane cost = profit!
profit = money_recieved_from_tickets - plane_cost
profit

800

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

## Combining conditions with logical operators

>The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

>The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

>The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [32]:
#Uploading the notebook :)
!pip install jovian --upgrade --quiet

In [33]:
import jovian

<IPython.core.display.Javascript object>

In [ ]:
jovian.commit(project='lecture-1-python-fundamentals')

<IPython.core.display.Javascript object>

[jovian] Please enter your API key ( from https://jovian.ai/ ):
API KEY: